libcodechecker/analyze/store_handler.py

```diff
@@ -0,0 +1,526 @@
+            .delete()
+
+        session.query(BugReportPoint) \
+            .filter(BugReportPoint.report_id == report_id) \
+            .delete()
+
+        LOG.debug("Add new bug path and events for report" +
+                  str(report_id))
+
+        LOG.debug("storing bug path")
+        store_bug_path(session, bugpath, report_id)
+        LOG.debug("storing events")
+        store_bug_events(session, events, report_id)
+
+    except Exception as ex:
+        raise shared.ttypes.RequestFailed(
+            shared.ttypes.ErrorCode.GENERAL,
+            str(ex))
+
+
+def is_same_event_path(report_id, events, session):
+    """
+    Checks if the given event path is the same as the one in the
+    events argument.
+    """
+    try:
+        q = session.query(BugPathEvent) \
+            .filter(BugPathEvent.report_id == report_id) \
+            .order_by(BugPathEvent.order)
+
+        for i, point2 in enumerate(q):
+            if i == len(events):
+                return False
+
+            point1 = events[i]
+
+            if point1.startCol != point2.col_begin or \
+                    point1.endCol != point2.col_end or \
+                    point1.msg != point2.msg:
+                return False
+
+        return True
+
+    except Exception as ex:
+        raise shared.ttypes.RequestFailed(
+            shared.ttypes.ErrorCode.GENERAL,
+            str(ex))
+
+
+def storeReportInfo(session,
+                    run_id,
+                    file_id,
+                    bug_hash,
+                    msg,
+                    bugpath,
+                    events,
+                    checker_id,
+                    checker_cat,
+                    bug_type,
+                    severity,
+                    detection_status='new'):
+    """
+    """
+    try:
+        LOG.debug("initializing report")
+        report = Report(run_id,
+                        bug_hash,
+                        file_id,
+                        msg,
+                        checker_id,
+                        checker_cat,
+                        bug_type,
+                        severity,
+                        detection_status)
+
+        session.add(report)
+        session.flush()
+
+        LOG.debug("storing bug path")
+        store_bug_path(session, bugpath, report.id)
+        LOG.debug("storing events")
+        store_bug_events(session, events, report.id)
+
+        return report.id
+    except Exception as ex:
+        raise shared.ttypes.RequestFailed(
+            shared.ttypes.ErrorCode.GENERAL,
+            str(ex))
+
+
+def addCheckerRun(session, storage_session, command, name, version, force):
+    """
+    Store checker run related data to the database.
+    By default updates the results if name already exists.
+    Using the force flag removes existing analysis results for a run.
+    """
+    try:
+        LOG.debug("adding checker run")
+
+        run = session.query(Run).filter(Run.name == name).first()
```

Comment:
Shouldn't we use `one_or_none` to enforce that there are no multiple runs with the same name?